### PR TITLE
Added DOC to Markdown Converter Function - Issue #23

### PR DESCRIPTION
**Pull request by aviral-bhardwaj**
### Added DOC to Markdown Converter Function

1. This commit introduces a new DocConverter class that extends the HtmlConverter class to handle the conversion of DOC files to Markdown format. The key features of this implementation are:
2. File Type Validation: The converter checks if the file extension is ".doc" before proceeding with the conversion.
3. Conversion Process:
4. The function uses the mammoth library to convert the DOC file to HTML.
5. It then utilizes the inherited _convert method to transform the HTML content into Markdown.

Issue URL- [This is Issue URL](https://github.com/microsoft/markitdown/issues/23) 

Closed Previous same PR due to the PyCharm Indentation Errors

### CODE REVIEW

**Comment by aviral-bhardwaj** [comment]
@StanFromIreland Please review this due to pycharm i was getting indentation error in my last PR

**Comment by realrajaryan** [comment]
should add test(s)

**Comment by aviral-bhardwaj** [comment]
adding but having problem to run that

**Comment by aviral-bhardwaj** [comment]
added test cases and it is passed

**Comment by aviral-bhardwaj** [comment]
can anyone please help me here @gagb , I am not able to run tests and pre-commits

**Comment by aviral-bhardwaj** [comment]
it is showing this in my pycharm 


================== 2 passed, 1 skipped, 3 warnings in 26.27s ===================
PASSED                        [ 33%]PASSED                         [ 66%]SKIPPED (do not run if
exiftool is not installed)                                               [100%]
Skipped: do not run if exiftool is not installed

**Review by l-lumin — CHANGES_REQUESTED** [review]

File: src/markitdown/_markitdown.py [hunk 1]
```diff
@@ -498,6 +498,30 @@ def convert(self, local_path, **kwargs) -> Union[None, DocumentConverterResult]:
...
+    """
+
+    def convert(self, local_path, **kwargs) -> Union[None, DocumentConverterResult]:
+        """
+               Args:
+                   local_path (str): The path to the DOC file.
+                   **kwargs: Additional arguments that may include 'file_extension'.
+               Returns:
+                   Union[None, DocumentConverterResult]: The conversion result or None if the file is not a DOC.
+               """
```
[the remark this answers is not in the document]
Comment (aviral-bhardwaj) [reply src/markitdown/_markitdown.py]: fixed

[the remark this answers is not in the document]
Comment (l-lumin) [reply src/markitdown/_markitdown.py]: can you check again? It still misaligned

**Comment by aviral-bhardwaj** [comment]
fixed according to review as the Args are creating issue in pychram so removed them according to all other functions

**Review by l-lumin — APPROVED** [review]
LGTM. Sorry, I'm just a random person passing by, so I can't magically approve this!

**Comment by aviral-bhardwaj** [comment]
@realrajaryan sir please can you help me to move forward

**Comment by gagb** [comment]
> can anyone please help me here @gagb , I am not able to run tests and pre-commits 

You can run precommit via 'pre-commit run --all-files' and then update the PR

**Comment by gagb** [comment]
Another PR introduced conflicts, can you please resolve?

**Comment by aviral-bhardwaj** [comment]
@gagb sir please review this , I have updated the test code ,help me to merge this

**Comment by gagb** [comment]
> @gagb sir please review this , I have updated the test code ,help me to merge this

Thanks, will get it merged soon. Appreciate your contributions.

**Comment by tungsten106** [comment]
@aviral-bhardwaj I wonder how you used mammoth to convert .doc file to html? I have tried on my own .doc file, it riased error `OSError: Could not find main document part. Are you sure this is a valid .docx file?`

File: src/markitdown/_markitdown.py [hunk 2]
```diff
@@ -712,6 +712,25 @@ def convert(self, local_path, **kwargs) -> Union[None, DocumentConverterResult]:
...
+    """
+
+    def convert(self, local_path, **kwargs) -> Union[None, DocumentConverterResult]:
+        extension = kwargs.get("file_extension", "")
+        if extension.lower() != ".doc":
+            return None
+
+        result = None
+        with open(local_path, "rb") as doc_file:
+            result = mammoth.convert_to_html(doc_file)
```
Comment (realrajaryan) [inline src/markitdown/_markitdown.py:727]: mammoth does not seem to support **.doc** files

File: tests/test_files/test.doc
Comment (realrajaryan) [inline tests/test_files/test.doc:1]: Looks like this is not a proper doc file. If you just renamed test.docx to test.doc, it does not effectively convert it into a .doc file, it merely changes the file extension while retaining the docx file format. This is why your test passes when it shouldn’t, as mammoth does not support doc files.

**Review by realrajaryan — CHANGES_REQUESTED** [review]
Mammoth does not support converting doc file format, need to find an alternative.

File: tests/test_files/test.doc
Comment (aviral-bhardwaj) [reply tests/test_files/test.doc:1]: ok checking , will update

**Comment by aviral-bhardwaj** [comment]
added new doc file this is running in my local without any issue 

Screenshot 2024-12-18 at 1 11 14 PM

File: tests/test_files/test.doc
Comment (realrajaryan) [reply tests/test_files/test.doc:1]: This is still not a valid doc file. The MIME type for this file is **application/octet-stream**, but for a doc file it’s supposed to be **application/msword**.

**Review by realrajaryan — CHANGES_REQUESTED** [review]

**Comment by aviral-bhardwaj** [comment]
added new doc file attaching its info snapshot also 

Screenshot 2024-12-18 at 1 49 03 PM

**Comment by aviral-bhardwaj** [comment]
yes doing more research on this , will update soon

**Comment by aviral-bhardwaj** [comment]
this PR can be closed as this feature is completed from this PR https://github.com/microsoft/markitdown/pull/153

**Pull request closed without merging**